projects/eslint-plugin/rules/general/lib/rules/no-conditional-object-keys.js

```diff
@@ -0,0 +1,72 @@
+/**
+ * SPDX-FileCopyrightText: © 2017 Liferay, Inc. <https://liferay.com>
+ * SPDX-License-Identifier: MIT
+ */
+
+const OBJECT = 'Object';
+const KEYS = 'keys';
+
+const message = "don't use truthy values in conditionals";
+
+module.exports = {
+	create(context) {
+		const throwError = (node, objectName, propertyName) => {
+			if (objectName === OBJECT && propertyName === KEYS) {
+				context.report({
+					message,
+					node,
+				});
+			}
+		};
+
+		return {
+			IfStatement(node) {
```

Comment:
Rather than checking just against `if ()` blocks, we also need to check against other types of conditionals.

For example:
```js
!Object.keys({}) ? false : true

!Object.keys({}) && 'test'
```


However, this also makes me think, do we ever have a case where we would want to allow `!Object.keys(someObj)` on it's own? I don't think so since that will always evaluate to false. So I think a more robust solution would be to check against all cases of `!Object.keys(someObj)`. We wouldn't want to check against cases that  call `.length` or some other key/method on `Object.keys(someObj)`.